### PR TITLE
fix: L0 segment has been loaded to worker during channel balance

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #37703

### CODE REVIEW

**Comment by weiliu1031** [comment]
rerun cpp-unit-test

File: internal/querynodev2/delegator/delegator_data.go [hunk 1]
```diff
@@ -412,9 +412,16 @@ func (sd *shardDelegator) LoadSegments(ctx context.Context, req *querypb.LoadSeg
 	log := sd.getLogger(ctx)
 
 	targetNodeID := req.GetDstNodeID()
+	if len(req.GetInfos()) > 0 && req.GetInfos()[0].Level == datapb.SegmentLevel_L0 {
+		// force l0 segment to load on delegator
+		if targetNodeID != paramtable.GetNodeID() {
+			targetNodeID = paramtable.GetNodeID()
+			log.Debug("unexpected L0 segment load on non-delegator node, force to load on delegator", zap.Int64("nodeIDInReq", req.GetDstNodeID()))
```
Comment (congqixia) [inline internal/querynodev2/delegator/delegator_data.go:419]: Maybe info or above?

**Pull request merged**